### ISSUE TITLE
🐛 Source Shopify: `KeyError` during normalization

### ISSUE THREAD

**Issue by GillParm** [airbytehq/airbyte#9976]
## Environment
- **Airbyte version**:  0.35.9-alpha
- **OS Version / Instance**: Ubuntu 18.04
- **Deployment**:  Docker
- **Source Connector and version**:  shopify 0.1.30
- **Destination Connector and version**:  BigQuery 0.6.5
- **Severity**: Medium 
- **Step where error happened**:  Sync job - basic normalization

## Current Behavior
at the basic normalization step, encounter a "KeyError: 'id'"

## Expected Behavior
should normalize

## Logs
Log below

<details>
<summary>LOG</summary> 

```

2022-02-01 08:55:51 [42mnormalization[0m > Traceback (most recent call last):
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/bin/transform-catalog", line 8, in <module>
2022-02-01 08:55:51 [42mnormalization[0m >     sys.exit(main())
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/transform.py", line 82, in main
2022-02-01 08:55:51 [42mnormalization[0m >     TransformCatalog().run(args)
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/transform.py", line 35, in run
2022-02-01 08:55:51 [42mnormalization[0m >     self.process_catalog()
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/transform.py", line 62, in process_catalog
2022-02-01 08:55:51 [42mnormalization[0m >     processor.process(catalog_file=catalog_file, json_column_name=json_col, default_schema=schema)
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/catalog_processor.py", line 74, in process
2022-02-01 08:55:51 [42mnormalization[0m >     nested_processors = stream_processor.process()
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 282, in process
2022-02-01 08:55:51 [42mnormalization[0m >     self.generate_scd_type_2_model(from_table, column_names),
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 750, in generate_scd_type_2_model
2022-02-01 08:55:51 [42mnormalization[0m >     "primary_key_partition": self.get_primary_key_partition(column_names),
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 953, in get_primary_key_partition
2022-02-01 08:55:51 [42mnormalization[0m >     return [self.get_primary_key_from_path(column_names, path) for path in self.primary_key]
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 953, in <listcomp>
2022-02-01 08:55:51 [42mnormalization[0m >     return [self.get_primary_key_from_path(column_names, path) for path in self.primary_key]
2022-02-01 08:55:51 [42mnormalization[0m >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 961, in get_primary_key_from_path
2022-02-01 08:55:51 [42mnormalization[0m >     if "type" in self.properties[field]:
2022-02-01 08:55:51 [42mnormalization[0m > KeyError: 'id'

```

</details>

## Steps to Reproduce
1. source shopify, destination BigQuery (set up with GCS staging)
2. sync with basic normalization

**Comment by mattbdc** [comment]
Seeing same. 

2022-01-27 18:14:25 normalization >     return [self.get_primary_key_from_path(column_names, path) for path in self.primary_key]
2022-01-27 18:14:25 normalization >   Generating airbyte_ctes/acmeinccorp/cw_inventory_items_ab2.sql from cw_inventory_items
2022-01-27 18:14:25 normalization >   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 961, in get_primary_key_from_path
2022-01-27 18:14:25 normalization >   Generating airbyte_views/acmeinccorp/cw_inventory_items_stg.sql from cw_inventory_items
2022-01-27 18:14:25 normalization >     if "type" in self.properties[field]:
2022-01-27 18:14:25 normalization > KeyError: 'id'
2022-01-27 18:14:25 normalization >   Generating airbyte_incremental/scd/acmeinccorp/cw_inventory_items_scd.sql from cw_inventory_items
2022-01-27 18:14:25 normalization >   Generating airbyte_incremental/acmeinccorp/cw_inventory_items.sql from cw_inventory_items
2022-01-27 18:14:25 normalization >   Generating airbyte_ctes/acmeinccorp/cw_inventory_levels_ab1.sql from cw_inventory_levels
2022-01-27 18:14:25 normalization >   Generating airbyte_ctes/acmeinccorp/cw_inventory_levels_ab2.sql from cw_inventory_levels
2022-01-27 18:14:25 normalization >   Generating airbyte_views/acmeinccorp/cw_inventory_levels_stg.sql from cw_inventory_levels
2022-01-27 18:14:25 normalization >

**Comment by ChristopheDuong** [comment]
Whoever takes on this issue should check if the `id` column declared as the primary key is actually part of the columns declared by the offending stream from that source.

I've seen sources declare some internal primary keys without publishing the column, so normalization wouldn't be able to actually use that primary key if not present in the data.

**Comment by bazarnov** [comment]
This is fixed in `0.1.31` version of Source Shopify, please update the connector to the latest `0.1.33` and try again.

**Comment by bazarnov** [comment]
Fixed in https://github.com/airbytehq/airbyte/pull/10175
